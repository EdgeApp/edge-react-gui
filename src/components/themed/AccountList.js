// @flow

import { type EdgeUserInfo } from 'edge-core-js'
import * as React from 'react'
import { Pressable, ScrollView, TouchableHighlight, View } from 'react-native'
import Animated, { Easing, useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated'
import Feather from 'react-native-vector-icons/Feather'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons'
import { sprintf } from 'sprintf-js'

import { deleteLocalAccount } from '../../actions/AccountActions.js'
import { logoutRequest } from '../../actions/LoginActions.js'
import { Fontello } from '../../assets/vector'
import s from '../../locales/strings'
import { useEffect, useState } from '../../types/reactHooks'
import { useDispatch, useSelector } from '../../types/reactRedux'
import { ButtonsModal } from '../modals/ButtonsModal.js'
import { Airship } from '../services/AirshipInstance.js'
import { type Theme, cacheStyles, useTheme } from '../services/ThemeContext'
import { DividerLine } from '../themed/DividerLine'
import { EdgeText } from '../themed/EdgeText'

type Props = {
  onPress: () => void,
  isOpen: boolean,
  duration: number
}

export function AccountList(props: Props) {
  const { isOpen, onPress, duration = 500 } = props
  const activeUsername = useSelector(state => state.core.account.username)
  const context = useSelector(state => state.core.context)
  const dispatch = useDispatch()
  const theme = useTheme()
  const styles = getStyles(theme)

  // Maintain the list of usernames:
  const [usernames, setUsernames] = useState(arrangeUsers(context.localUsers, activeUsername))
  useEffect(() => context.watch('localUsers', localUsers => setUsernames(arrangeUsers(localUsers, activeUsername))), [activeUsername, context])

  const handleDelete = (username: string) => {
    Airship.show(bridge => (
      <ButtonsModal
        bridge={bridge}
        title={s.strings.delete_account_header}
        message={sprintf(s.strings.delete_username_account, username)}
        buttons={{
          ok: {
            label: s.strings.string_delete,
            async onPress() {
              await dispatch(deleteLocalAccount(username))
              height.value -= styles.row.height
              return true
            }
          },
          cancel: { label: s.strings.string_cancel }
        }}
      />
    ))
  }

  const handleSwitchAccount = (username: string) => {
    dispatch(logoutRequest(username))
  }

  const listHeight = styles.row.height * usernames.length + theme.rem(1)
  const height = useSharedValue(isOpen ? listHeight : 0)
  useEffect(() => {
    height.value = isOpen ? listHeight : 0
  }, [height, isOpen, listHeight])

  const animatedStyle = useAnimatedStyle(() => {
    return {
      height: withTiming(height.value, {
        duration: duration,
        easing: Easing.out(Easing.exp)
      })
    }
  })

  return (
    <View>
      <Pressable onPress={onPress}>
        <View style={styles.header}>
          <Fontello name="edge.logo" style={styles.iconUser} size={theme.rem(1.5)} color={theme.mainMenuIcon} />
          <EdgeText style={styles.text}>{activeUsername}</EdgeText>
          <Feather name="chevron-down" color={theme.mainMenuIcon} size={theme.rem(1.5)} />
        </View>
      </Pressable>
      <DividerLine marginRem={[1, -2, 0, 0]} />
      <Animated.View style={[styles.root, animatedStyle]}>
        <ScrollView>
          {usernames.map((username: string) => (
            <View key={username} style={styles.row}>
              <TouchableHighlight onPress={() => handleSwitchAccount(username)}>
                <EdgeText style={styles.text}>{username}</EdgeText>
              </TouchableHighlight>
              <TouchableHighlight onPress={() => handleDelete(username)}>
                <View>
                  <MaterialIcon size={theme.rem(1.5)} name="close" color={theme.mainMenuIcon} />
                </View>
              </TouchableHighlight>
            </View>
          ))}
        </ScrollView>
      </Animated.View>
    </View>
  )
}

/**
 * Given a list of users from the core,
 * remove the given user, then organize the 3 most recent users,
 * followed by the rest in alphabetical order.
 */
function arrangeUsers(localUsers: EdgeUserInfo[], activeUsername: string): string[] {
  // Sort the users according to their last login date:
  const usernames = localUsers
    .filter(info => info.username !== activeUsername)
    .sort((a, b) => {
      const { lastLogin: aDate = new Date(0) } = a
      const { lastLogin: bDate = new Date(0) } = b
      return aDate.valueOf() - bDate.valueOf()
    })
    .map(info => info.username)

  // Sort everything after the last 3 entries alphabetically:
  const oldUsernames = usernames.slice(3).sort((a: string, b: string) => {
    const stringA = a.toUpperCase()
    const stringB = b.toUpperCase()
    if (stringA < stringB) return -1
    if (stringA > stringB) return 1
    return 0
  })

  return [...usernames.slice(0, 3), ...oldUsernames]
}

const getStyles = cacheStyles((theme: Theme) => ({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: theme.rem(2.5),
    marginLeft: theme.rem(2)
  },
  text: {
    marginRight: 'auto',
    fontFamily: theme.fontFaceBold
  },
  header: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    height: theme.rem(2)
  },
  iconUser: {
    marginRight: theme.rem(1.5)
  }
}))
