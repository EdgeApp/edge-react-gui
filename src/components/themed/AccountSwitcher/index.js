// @flow

export * from './helpers'
export { default as SwitcherHeader } from './SwitcherHeader'
export { default as SwitcherList } from './SwitcherList'
