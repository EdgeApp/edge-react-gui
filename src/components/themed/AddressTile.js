// @flow

import Clipboard from '@react-native-community/clipboard'
import type { EdgeCurrencyConfig, EdgeCurrencyWallet, EdgeParsedUri } from 'edge-core-js'
import * as React from 'react'
import { AppState, TouchableOpacity, View } from 'react-native'
import FontAwesome from 'react-native-vector-icons/FontAwesome'
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5'

import { CURRENCY_PLUGIN_NAMES } from '../../constants/WalletAndCurrencyConstants'
import s from '../../locales/strings.js'
import { checkPubAddress } from '../../modules/FioAddress/util'
import { connect } from '../../types/reactRedux.js'
import { type GuiMakeSpendInfo } from '../../types/types.js'
import { AddressModal } from '../modals/AddressModal'
import { paymentProtocolUriReceived } from '../modals/paymentProtocolUriReceived.js'
import { ScanModal } from '../modals/ScanModal.js'
import { shouldContinueLegacy } from '../modals/shouldContinueLegacy.js'
import { Airship, showError } from '../services/AirshipInstance'
import { type Theme, type ThemeProps, cacheStyles, withTheme } from '../services/ThemeContext.js'
import { EdgeText } from './EdgeText'
import { Tile } from './Tile.js'

type OwnProps = {
  coreWallet: EdgeCurrencyWallet,
  currencyCode: string,
  title: string,
  recipientAddress: string,
  onChangeAddress: (guiMakeSpendInfo: GuiMakeSpendInfo, parsedUri?: EdgeParsedUri) => Promise<void>,
  resetSendTransaction: () => void,
  lockInputs?: boolean,
  addressTileRef: any,
  isCameraOpen: boolean,
  fioToAddress?: string
}
type StateProps = {
  fioPlugin?: EdgeCurrencyConfig
}
type State = {
  clipboard: string,
  loading: boolean
}
type Props = OwnProps & StateProps & ThemeProps

const isLegacyAddressUri = (parsedUri: EdgeParsedUri): boolean => {
  return !!parsedUri.legacyAddress
}

const isPaymentProtocolUri = (parsedUri: EdgeParsedUri): boolean => {
  // $FlowFixMe should be paymentProtocolUrl (lowercased)?
  return !!parsedUri.paymentProtocolURL && !parsedUri.publicAddress
}

class AddressTileComponent extends React.PureComponent<Props, State> {
  constructor(props: Props) {
    super(props)

    this.state = {
      clipboard: '',
      loading: false
    }
  }

  componentDidMount(): void {
    AppState.addEventListener('change', this.handleAppStateChange)

    this._setClipboard(this.props)
    this.props.addressTileRef(this)
    if (this.props.isCameraOpen) {
      this.handleScan()
    }
  }

  componentWillUnmount(): void {
    AppState.removeEventListener('change', this.handleAppStateChange)

    this.props.addressTileRef(undefined)
  }

  componentDidUpdate(prevProps: Props) {
    if (this.props.isCameraOpen && !prevProps.isCameraOpen) {
      this.handleScan()
    }
  }

  handleAppStateChange = appState => {
    if (appState === 'active') this._setClipboard(this.props)
  }

  reset() {
    this._setClipboard(this.props)
    this.props.resetSendTransaction()
  }

  onChangeAddress = async (address: string) => {
    if (!address) return
    const { onChangeAddress, coreWallet, currencyCode, fioPlugin } = this.props

    this.setState({ loading: true })
    let fioAddress
    if (fioPlugin) {
      try {
        const publicAddress = await checkPubAddress(fioPlugin, address.toLowerCase(), coreWallet.currencyInfo.currencyCode, currencyCode)
        fioAddress = address.toLowerCase()
        address = publicAddress
      } catch (e) {
        if (!e.code || e.code !== fioPlugin.currencyInfo.defaultSettings.errorCodes.INVALID_FIO_ADDRESS) {
          this.setState({ loading: false })
          return showError(e)
        }
      }
    }
    try {
      const parsedUri: EdgeParsedUri & { paymentProtocolURL?: string } = await coreWallet.parseUri(address, currencyCode)

      this.setState({ loading: false })

      if (isLegacyAddressUri(parsedUri)) {
        if (!(await shouldContinueLegacy())) return
      }

      // Missing isPrivateKeyUri Modal

      if (isPaymentProtocolUri(parsedUri)) {
        const guiMakeSpendInfo: ?GuiMakeSpendInfo = await paymentProtocolUriReceived(parsedUri, coreWallet)

        if (guiMakeSpendInfo) {
          onChangeAddress(guiMakeSpendInfo)
        }

        return
      }

      if (!parsedUri.publicAddress) {
        return showError(s.strings.scan_invalid_address_error_title)
      }

      // set address
      onChangeAddress({ fioAddress, isSendUsingFioAddress: !!fioAddress }, parsedUri)
    } catch (e) {
      showError(`${s.strings.scan_invalid_address_error_title} ${s.strings.scan_invalid_address_error_description}`)
      this.setState({ loading: false })
    }
  }

  _setClipboard = async props => {
    const { coreWallet, currencyCode } = props

    try {
      this.setState({ loading: true })
      const uri = await Clipboard.getString()

      // Will throw in case uri is invalid
      await coreWallet.parseUri(uri, currencyCode)

      this.setState({
        clipboard: uri,
        loading: false
      })
    } catch (e) {
      this.setState({ loading: false, clipboard: '' })
      // Failure is acceptable
    }
  }

  handlePasteFromClipboard = () => {
    const { clipboard } = this.state
    this.onChangeAddress(clipboard)
  }

  handleScan = () => {
    Airship.show(bridge => <ScanModal bridge={bridge} title={s.strings.scan_qr_label} />)
      .then((result: string | void) => {
        if (result) {
          this.onChangeAddress(result)
        }
      })
      .catch(error => {
        showError(error)
      })
  }

  handleChangeAddress = async () => {
    const { coreWallet, currencyCode } = this.props
    Airship.show(bridge => <AddressModal bridge={bridge} walletId={coreWallet.id} currencyCode={currencyCode} title={s.strings.scan_address_modal_title} />)
      .then((result: string | null) => {
        if (result) {
          this.onChangeAddress(result)
        }
      })
      .catch(error => {
        showError(error)
      })
  }

  handleTilePress = () => {
    const { lockInputs, recipientAddress } = this.props
    if (!lockInputs && !!recipientAddress) {
      this.reset()
    }
  }

  render() {
    const { fioToAddress, recipientAddress, lockInputs, theme, title } = this.props
    const { loading } = this.state
    const styles = getStyles(theme)
    const copyMessage = this.state.clipboard ? `${s.strings.string_paste}: ${this.state.clipboard}` : null
    const tileType = loading ? 'loading' : !!recipientAddress && !lockInputs ? 'touchable' : 'static'
    return (
      <View>
        <Tile type={tileType} title={title} onPress={this.handleTilePress}>
          {!recipientAddress && (
            <View style={styles.buttonsContainer}>
              <TouchableOpacity style={styles.buttonContainer} onPress={this.handleChangeAddress}>
                <FontAwesome name="edit" size={theme.rem(2)} color={theme.iconTappable} />
                <EdgeText style={styles.buttonText}>{s.strings.enter_as_in_enter_address_with_keyboard}</EdgeText>
              </TouchableOpacity>
              <TouchableOpacity style={styles.buttonContainer} onPress={this.handleScan}>
                <FontAwesome5 name="expand" size={theme.rem(2)} color={theme.iconTappable} />
                <EdgeText style={styles.buttonText}>{s.strings.scan_as_in_scan_barcode}</EdgeText>
              </TouchableOpacity>
              {copyMessage && (
                <TouchableOpacity style={styles.buttonContainer} onPress={this.handlePasteFromClipboard}>
                  <FontAwesome5 name="clipboard" size={theme.rem(2)} color={theme.iconTappable} />
                  <EdgeText style={styles.buttonText}>{s.strings.string_paste}</EdgeText>
                </TouchableOpacity>
              )}
            </View>
          )}
          {fioToAddress == null ? null : <EdgeText>{fioToAddress + '\n'}</EdgeText>}
          {recipientAddress == null ? null : (
            <EdgeText numberOfLines={3} disableFontScaling>
              {recipientAddress}
            </EdgeText>
          )}
        </Tile>
      </View>
    )
  }
}

const getStyles = cacheStyles((theme: Theme) => ({
  buttonsContainer: {
    paddingTop: theme.rem(1),
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center'
  },
  buttonText: {
    fontSize: theme.rem(0.75),
    color: theme.textLink
  }
}))

const AddressTileConnector = connect<StateProps, {}, OwnProps>(
  state => ({
    fioToAddress: state.ui.scenes.sendConfirmation.guiMakeSpendInfo?.fioAddress,
    fioPlugin: state.core.account.currencyConfig[CURRENCY_PLUGIN_NAMES.FIO]
  }),
  dispatch => ({})
)(withTheme(AddressTileComponent))

// $FlowFixMe - forwardRef is not recognize by flow?
export const AddressTile = React.forwardRef((props, ref) => <AddressTileConnector {...props} addressTileRef={ref} />) // eslint-disable-line
