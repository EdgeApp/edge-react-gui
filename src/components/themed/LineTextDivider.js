// @flow

import * as React from 'react'
import { View } from 'react-native'

import { type Theme, type ThemeProps, cacheStyles, withTheme } from '../services/ThemeContext.js'
import { EdgeText } from './EdgeText.js'

type Props = {
  title?: string,
  children?: React.Node,
  lowerCased?: boolean
}

class LineTextDividerComponent extends React.PureComponent<Props & ThemeProps> {
  render() {
    const { title, children, theme } = this.props
    const styles = getStyles(theme)
    return (
      <View style={styles.container}>
        <View style={styles.line} />
        {title ? <EdgeText style={[styles.title, this.props.lowerCased ? styles.lowerCase : null]}>{title}</EdgeText> : null}
        {children}
        <View style={styles.line} />
      </View>
    )
  }
}

const getStyles = cacheStyles((theme: Theme) => ({
  container: {
    width: '100%',
    paddingHorizontal: theme.rem(1),
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginVertical: theme.rem(1)
  },
  line: {
    flex: 1,
    borderBottomWidth: theme.thinLineWidth,
    borderBottomColor: theme.titleLineDivider
  },
  title: {
    fontSize: theme.rem(1),
    paddingHorizontal: theme.rem(0.75),
    fontFamily: theme.fontFaceMedium,
    color: theme.secondaryText
  },
  lowerCase: {
    textTransform: 'lowercase'
  }
}))

export const LineTextDivider = withTheme(LineTextDividerComponent)
