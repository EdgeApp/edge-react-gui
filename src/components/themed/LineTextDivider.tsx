import * as React from 'react'
import { View } from 'react-native'

import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
import { EdgeText } from './EdgeText'

interface Props {
  title?: string
  children?: React.ReactNode
  lowerCased?: boolean
}

export const LineTextDividerComponent = (props: Props) => {
  const { title, children, lowerCased } = props
  const theme = useTheme()
  const styles = getStyles(theme)
  return (
    <View style={styles.container}>
      <View style={styles.line} />
      {title ? <EdgeText style={[styles.title, lowerCased ? styles.lowerCase : null]}>{title}</EdgeText> : null}
      {children}
      <View style={styles.line} />
    </View>
  )
}

const getStyles = cacheStyles((theme: Theme) => ({
  container: {
    flexGrow: 1,
    paddingHorizontal: theme.rem(0.5),
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginVertical: theme.rem(0.5)
  },
  line: {
    flex: 1,
    borderBottomWidth: theme.thinLineWidth,
    borderBottomColor: theme.titleLineDivider
  },
  title: {
    fontSize: theme.rem(1),
    paddingHorizontal: theme.rem(0.75),
    fontFamily: theme.fontFaceMedium,
    color: theme.secondaryText
  },
  lowerCase: {
    textTransform: 'lowercase'
  }
}))

export const LineTextDivider = React.memo(LineTextDividerComponent)
