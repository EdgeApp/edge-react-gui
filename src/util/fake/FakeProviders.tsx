import * as React from 'react'
import { Metrics, SafeAreaProvider } from 'react-native-safe-area-context'
import { Provider } from 'react-redux'
import { applyMiddleware, createStore } from 'redux'
import thunk from 'redux-thunk'

import { rootReducer, RootState } from '../../reducers/RootReducer'

type DeepPartial<T> = T extends object
  ? {
      [P in keyof T]?: DeepPartial<T[P]>
    }
  : T

export type FakeState = DeepPartial<RootState>

interface Props {
  children: React.ReactNode
  initialState?: FakeState
}

export function FakeProviders(props: Props) {
  const { children, initialState = {} } = props

  const store = React.useMemo(() => createStore(rootReducer, initialState as any, applyMiddleware(thunk)), [initialState])
  return (
    <SafeAreaProvider initialMetrics={initialMetrics}>
      <Provider store={store}>{children}</Provider>
    </SafeAreaProvider>
  )
}

// These match what our old react-native-safe-area-view library returned:
const initialMetrics: Metrics = {
  frame: { height: 1334, width: 750, x: 0, y: 0 },
  insets: { bottom: 0, left: 0, right: 0, top: 20 }
}
