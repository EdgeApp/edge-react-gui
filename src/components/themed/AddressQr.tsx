import { EdgeCurrencyWallet, EdgeTokenId } from 'edge-core-js'
import * as React from 'react'
import { View } from 'react-native'

import { useAsyncValue } from '../../hooks/useAsyncValue'
import { getCurrencyCode } from '../../util/CurrencyInfoHelpers'
import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
import { QrCode } from '../themed/QrCode'

interface Props {
  address: string
  tokenId: EdgeTokenId
  wallet: EdgeCurrencyWallet

  nativeAmount?: string
  onPress?: (encodedUri?: string) => void
}

export const AddressQr = (props: Props) => {
  const theme = useTheme()
  const styles = getStyles(theme)
  const { address, tokenId, wallet, nativeAmount, onPress = () => {} } = props

  const [encodedUri] = useAsyncValue(
    async () => await wallet.encodeUri({ publicAddress: address, currencyCode: getCurrencyCode(wallet, tokenId), nativeAmount }),
    [address, tokenId, nativeAmount, wallet]
  )

  return (
    <View style={styles.container}>
      <QrCode data={encodedUri} tokenId={tokenId} wallet={wallet} onPress={() => onPress(encodedUri)} marginRem={0} />
    </View>
  )
}

const getStyles = cacheStyles((theme: Theme) => ({
  container: {
    aspectRatio: 1,
    height: '100%',
    width: '100%',
    alignSelf: 'center'
  }
}))
