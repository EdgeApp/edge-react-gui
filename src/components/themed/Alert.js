// @flow

import * as React from 'react'
import { Platform, StyleSheet, TouchableOpacity, View } from 'react-native'
import IonIcon from 'react-native-vector-icons/Ionicons'

import { fixSides, mapSides, sidesToMargin, sidesToPadding } from '../../util/sides.js'
import { type Theme, cacheStyles, useTheme } from '../services/ThemeContext.js'
import { EdgeText } from '../themed/EdgeText'

type AlerType = 'info' | 'warning' | 'error' | 'success'

type Props = {
  type: AlerType,
  title: string,
  message?: string,
  numberOfLines?: number,
  children?: React.Node,
  marginRem?: number[] | number,
  paddingRem?: number[] | number,
  onPress?: () => void
}

type StylesOptions = { typeColor: string }

// TODO: Implement alert colors such as info and others
export const useAlertTypeColor = (type: AlerType) => {
  const theme = useTheme()

  switch (type) {
    case 'warning':
      return theme.warningText
    case 'error':
      return theme.dangerText
    default:
      return theme.primaryText
  }
}

// TODO: Implement alert icons such as info and others
export const useAlertTypeIcon = (type: AlerType, color: string, size: number, style: StyleSheet.Styles) => {
  switch (type) {
    case 'warning':
    case 'error':
      return (
        <IonIcon style={style} name={Platform.OS === 'ios' ? 'ios-information-circle-outline' : 'md-information-circle-outline'} color={color} size={size} />
      )
    default:
      return null
  }
}

export function Alert({ type, title, message, numberOfLines = 2, marginRem, paddingRem, onPress, children }: Props) {
  const theme = useTheme()
  const margin = sidesToMargin(mapSides(fixSides(marginRem, 1), theme.rem))
  const padding = sidesToPadding(mapSides(fixSides(paddingRem, 1), theme.rem))

  const typeColor = useAlertTypeColor(type)

  const styles = getStyles(theme, { typeColor })

  const icon = useAlertTypeIcon(type, typeColor, theme.rem(1.25), styles.icon)

  const result = (
    <View style={[styles.alert, margin, padding]}>
      <View style={styles.header}>
        {icon}
        <EdgeText style={styles.title}>{title}</EdgeText>
      </View>
      {message && (
        <EdgeText style={styles.message} numberOfLines={numberOfLines}>
          {message}
        </EdgeText>
      )}
      {children}
    </View>
  )

  return onPress ? <TouchableOpacity onPress={onPress}>{result}</TouchableOpacity> : result
}

const getStyles = (theme: Theme, { typeColor }: StylesOptions) =>
  cacheStyles((theme: Theme) => ({
    alert: {
      borderWidth: theme.cardBorder,
      borderColor: typeColor,
      borderRadius: theme.cardBorderRadius,
      alignSelf: 'stretch'
    },
    header: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center'
    },
    title: {
      fontSize: theme.rem(0.75),
      color: typeColor
    },
    icon: {
      marginRight: theme.rem(0.5)
    },
    message: {
      fontSize: theme.rem(0.75),
      color: typeColor,
      fontFamily: theme.fontFaceDefault
    }
  }))(theme)
