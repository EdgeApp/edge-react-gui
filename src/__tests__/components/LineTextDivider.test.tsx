import { describe, expect, it } from '@jest/globals'
import { render } from '@testing-library/react-native'
import * as React from 'react'

import { LineTextDividerComponent } from '../../components/themed/LineTextDivider'
import { FakeProviders } from '../../util/fake/FakeProviders'

describe('LineTextDivider', () => {
  it('should render with loading props', () => {
    const fakeChild: React.ReactNode = 'hello'

    const rendered = render(
      <FakeProviders>
        <LineTextDividerComponent title="string" lowerCased>
          {fakeChild}
        </LineTextDividerComponent>
      </FakeProviders>
    )

    expect(rendered.toJSON()).toMatchSnapshot()
    rendered.unmount()
  })
})
