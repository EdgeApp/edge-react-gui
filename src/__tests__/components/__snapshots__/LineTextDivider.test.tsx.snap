// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineTextDivider should render with loading props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 22,
      "paddingHorizontal": 22,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#A4C7DF",
        "borderBottomWidth": 1,
        "flex": 1,
      }
    }
  />
  <WithTheme(EdgeTextComponent)
    style={
      Array [
        Object {
          "color": "#A4C7DF",
          "fontFamily": "Quicksand-Medium",
          "fontSize": 22,
          "paddingHorizontal": 17,
        },
        Object {
          "textTransform": "lowercase",
        },
      ]
    }
  >
    string
  </WithTheme(EdgeTextComponent)>
  hello
  <View
    style={
      Object {
        "borderBottomColor": "#A4C7DF",
        "borderBottomWidth": 1,
        "flex": 1,
      }
    }
  />
</View>
`;
